src/pages/index.js

```diff
@@ -0,0 +1,292 @@
+
+// Store the current selected card and its ID
+let selectedCard;
+let selectedCardId;
+
+api
+  .getAppInfo()
+  .then(([cardsData, userData]) => {
+    avatarImage.src = userData.avatar;
+    profileNameElement.textContent = userData.name;
+    profileDescriptionElement.textContent = userData.about;
+
+    cardsData.forEach((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.append(cardElement);
+    });
+  })
+  .catch(console.error);
+
+// Opens a modal
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscClose);
+}
+
+// Closes a modal
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+
+// Opens profile edit modal
+function openEditProfileModal() {
+  nameInput.value = profileNameElement.textContent;
+  descriptionInput.value = profileDescriptionElement.textContent;
+  resetValidation(profileForm, [nameInput, descriptionInput], settings);
+  openModal(editProfileModal);
+  disableButton(profileFormSubmitButton, settings);
+}
+
+// Opens new post modal
+function openNewPostModal() {
+  openModal(newPostModal);
+}
+
+// Opens avatar edit modal
+function openAvatarModal() {
+  resetValidation(avatarForm, [avatarLinkInput], settings);
+  openModal(avatarModal);
+  disableButton(avatarFormSubmitButton, settings);
+}
+
+// Submits profile form
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .editUserInfo({
+      name: nameInput.value,
+      about: descriptionInput.value,
+    })
+    .then((userInfo) => {
+      profileNameElement.textContent = userInfo.name;
+      profileDescriptionElement.textContent = userInfo.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+// Submits new post form
+function handleNewPostFormSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  const cardData = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  api
+    .addCard(cardData)
+    .then((newCardData) => {
+      const cardElement = getCardElement(newCardData);
+      cardList.prepend(cardElement);
+      evt.target.reset();
+      disableButton(cardFormSubmitButton, settings);
+      closeModal(newPostModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitButton, false);
+    });
+}
+
+// Submits avatar form
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api
+    .setUserAvatar({
+      avatar: avatarLinkInput.value,
+    })
+    .then((userData) => {
+      document.querySelector(".profile__image").src = userData.avatar;
```

Comment:
`document.querySelector(".profile__image")` has already been found in line 24 as `avatarImage`. Please, use it here


```js

      avatarImage.src = userData.avatar;

```